p2p/nat/nat_stun.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package nat
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pion/stun"
+)
+
+// The code are from erigon p2p/nat/nat_stun.go
+// This stun server is part of the mainnet infrastructure.
+// The addr are from https://github.com/ethereum/trin/blob/master/portalnet/src/socket.rs
+const STUNDefaultServerAddr = "159.223.0.83:3478"
+
+type STUN struct {
+	serverAddr string
+}
+
+func NewSTUN(serverAddr string) STUN {
+	if serverAddr == "" {
```

Comment:
```suggestion
	if serverAddr == "default" {
```
Imo better if using the default is an explicit opt-in from the user

Reply:
Should also verify that `serverAddr` can be converted into a valid `udp4` addr, and error otherwise. 
And why not let `serverAddr` be and `*UDPAddr` instead?
```golang
	addr, err := net.ResolveUDPAddr("udp4", serverAddr)
        if err != nil{
              return nil, err
        }
        return STUN{ serverAddr: addr}, nil
```

Reply:
@holiman 
for this one`Imo better if using the default is an explicit opt-in from the user`

you mean use should use like this `--nat stun: default`